src/test/java/io/strimzi/kafka/bridge/http/ProducerIT.java

```diff
@@ -1117,4 +1118,77 @@ void sendSimpleMessageWithWrongContentType(VertxTestContext context) throws Inte
+
+        final JsonObject messageKafkaJsonContentType = createRecordsToSend("Hello from the other side", "message-key", BridgeContentType.KAFKA_JSON_JSON);
+        final JsonObject messageKafkaJsonBinaryContentType = createRecordsToSend("message-bin-key", "content....----", BridgeContentType.KAFKA_JSON_BINARY);
+
+        future.get();
+
+        producerService()
+            .sendRecordsRequest(topic, messageKafkaJsonContentType, BridgeContentType.KAFKA_JSON_JSON)
+            .sendJsonObject(messageKafkaJsonContentType, response -> handleResponse(context, response, BridgeContentType.KAFKA_JSON_JSON));
+
+        // Second request with binary content, simulating a binary message
+        producerService()
+            .sendRecordsRequest(topic, messageKafkaJsonBinaryContentType, BridgeContentType.KAFKA_JSON_BINARY)
+            .sendBuffer(Buffer.buffer(messageKafkaJsonBinaryContentType.toString()), response -> {
+                handleResponse(context, response, BridgeContentType.KAFKA_JSON_BINARY);
+                context.completeNow();
+            });
+
+        assertThat(context.awaitCompletion(TEST_TIMEOUT, TimeUnit.SECONDS), is(true));
+    }
+
+    /**
+     * Creates a JSON object representing records to send. Depending on the content type, the method adjusts the key and value
+     * by encoding them in base64 format for binary content.
+     *
+     * @param key           The key of the message
+     * @param value         The value of the message
+     * @param contentType   The content type of the message, determining whether the message is treated as JSON or binary.
+     * @return A JsonObject representing the records to send, formatted according to the specified content type.
+     */
+    private JsonObject createRecordsToSend(String key, String value, final String contentType) {
+        if (BridgeContentType.KAFKA_JSON_BINARY.equals(contentType)) {
+            key = DatatypeConverter.printBase64Binary(key.getBytes());
+            value = DatatypeConverter.printBase64Binary(value.getBytes());
+        }
+
+        JsonObject json = new JsonObject()
+            .put("key", key)
+            .put("value", value);
+
+        JsonArray records = new JsonArray().add(json);
+
+        return new JsonObject().put("records", records);
+    }
+
+    /**
+     * Handles the response from sending a record to Kafka, logging the outcome and verifying the HTTP status code.
+     * This method is used as a callback for the asynchronous HTTP request to send records to Kafka.
+     *
+     * @param context       The VertxTestContext used for assertions and to mark the test as completed.
+     * @param ar            The result of the asynchronous HTTP request, containing either the response or an error.
+     * @param contentType   The content type of the message that was sent, used for logging purposes.
+     */
+    private void handleResponse(final VertxTestContext context, final AsyncResult<HttpResponse<JsonObject>> ar,
```

Comment:
Well, I don't want to refactor the whole test suite. I didn't want to make redundant code. Agree with changing the signature of the method (i.e., adding status code as additional parameter). I could create an issue to refactor it with such auxiliary method.

Reply:
Tbh if you are planning to refactor all tests right after this PR I am ok, otherwise I am for removing this method and just duplicate the code as we have all across the other tests. Otherwise it will end to have this method here for months without refactoring all the other tests, there will be no value in the method in the end.